tools/cbmc/include/cbmc.h

```diff
@@ -25,3 +25,26 @@ enum CBMC_LOOP_CONDITION { CBMC_LOOP_BREAK, CBMC_LOOP_CONTINUE, CBMC_LOOP_RETURN
 
 #define __CPROVER_printf_ptr(var) { uint8_t *ValueOf_ ## var = (uint8_t *) var; }
 #define __CPROVER_printf2_ptr(str,exp) { uint8_t *ValueOf_ ## str = (uint8_t *) (exp); }
+
+/* CBMC assert to test pvPortMalloc result when xWantedSize is 0. Mostly used to report
+ * full coverage on pvPortMalloc, but use with caution as it might complicate debugging
+ */
+#define __CPROVER_assert_zero_allocation() __CPROVER_assert( pvPortMalloc(0) == NULL, "pvPortMalloc allows zero-allocated memory.")
```

Comment:
I reviewed these in some dependent PRs.  
1. The macro name only makes sense once I read the comment. It should be self-explanatory
1. Why does this lead to full coverage? That is not clear. How does it complicate debugging?
1. The string on the assert itself is incorrect.  its allowed under the description of " If size is 0, either a null pointer or a unique pointer that can be successfully passed to free() shall be returned." 
https://pubs.opengroup.org/onlinepubs/009695399/functions/malloc.html

Reply:
This seems to be necessary because of the implementation of `pvPortMalloc` below, which always returns `NULL` when passed `0`.

I am going to remove that case from `pvPortMalloc`, so there should be no need for this macro at all. I'll delete from this PR and from any PRs that use it.